apiserver/facades/agent/uniter/uniter_test.go

```diff
@@ -713,13 +767,7 @@ func (s *uniterSuite) TestGetPrincipal(c *gc.C) {
 	// Now try as subordinate's agent.
 	subAuthorizer := s.authorizer
 	subAuthorizer.Tag = subordinate.Tag()
-	subUniter, err := uniter.NewUniterAPI(facadetest.Context{
-		State_:             s.State,
-		Resources_:         s.resources,
-		Auth_:              subAuthorizer,
-		LeadershipChecker_: s.State.LeadershipChecker(),
-	})
-	c.Assert(err, jc.ErrorIsNil)
+	subUniter := s.newUniterAPI(c, s.State, subAuthorizer)
```

Comment:
Is not making this same change at line 251 deliberate?

Reply:
Yes; we are check the error there, rather than setting a working API.